dev-docs/003-io-module-rfc.md

```diff
@@ -0,0 +1,145 @@
+- Start Date: 2019-04-29
+- RFC PR: [#435](https://github.com/uber/xviz/pull/435)
+- XVIZ Issue: [#434](https://github.com/uber/xviz/issues/434)
+
+# Summary
```

Comment:
So what I have been doing in the internal XVIZ implementation is heading this way:
 - **encoder** - code to go from C++ types to JSON, GLB, and eventually protobuf XVIZ structures
 - **server** - code that helps serve XVIZ requests, uses the **encode**
 - **builder** - contains just the user facing API
 - **schema** - code to load the JSON schema files
 - **validator** - code to validate XVIZ content

Reply:
Does your encoder got from XVIZ => C++ also?  That would be more a decoder.  Are you suggesting to change from **io reader** and **io writer** to **encoder** and **decoder** ?  where **io** is just the containing module since these two actions are closely related.

Reply:
You made a comment that your ok with the reader/writer & io  vs encoder right.